pkg/runtime/engine/config/parser.go

```diff
@@ -17,147 +17,156 @@ import (
-			var err error
-
-			if directives[dir] != nil && directives[dir][0] != "" {
-				n, err = strconv.ParseInt(directives[dir][0], 0, 64)
-			} else {
-				n, err = strconv.ParseInt(def, 0, 64)
+			if !found && len(authorized) > 0 {
+				return fmt.Errorf("value authorized for directive %q are %s", dir, authorized)
 			}
+			valueField.SetBool(value[0] == "yes")
+		case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+			n, err := strconv.ParseInt(value[0], 0, 64)
 			if err != nil {
 				return err
 			}
 			valueField.SetInt(n)
 		case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
-			var n uint64
-			var err error
-
-			if directives[dir] != nil && directives[dir][0] != "" {
-				n, err = strconv.ParseUint(directives[dir][0], 0, 64)
-			} else {
-				n, err = strconv.ParseUint(def, 0, 64)
-			}
+			n, err := strconv.ParseUint(value[0], 0, 64)
 			if err != nil {
 				return err
 			}
 			valueField.SetUint(n)
 		case reflect.String:
+			if len(value) == 0 {
+				value = []string{""}
+			}
 			found := false
-			if directives[dir] != nil {
-				// To allow for string fields which are intended to be set to *any* value, we must
-				// handle the case where authorized isn't set (implies any value is acceptable)
-				if len(authorized) == 1 && authorized[0] == "" {
-					valueField.SetString(directives[dir][0])
+			for _, a := range authorized {
+				if a == value[0] {
 					found = true
-				} else {
-					for _, a := range authorized {
-						if a == directives[dir][0] {
-							valueField.SetString(a)
-							found = true
-							break
-						}
-					}
+					break
 				}
-				if !found {
-					return fmt.Errorf("value authorized for directive '%s' are %s", dir, authorized)
-				}
-			} else {
-				valueField.SetString(def)
 			}
+			if !found && len(authorized) > 0 && value[0] != "" {
+				return fmt.Errorf("value authorized for directive '%s' are %s", dir, authorized)
+			}
+			valueField.SetString(strings.Join(value, ","))
 		case reflect.Slice:
-			l := len(directives[dir])
-			switch valueField.Interface().(type) {
+			l := len(value)
+			v := reflect.MakeSlice(typeField.Type, l, l)
+			valueField.Set(v)
+
+			switch t := valueField.Interface().(type) {
 			case []string:
-				if l == 1 {
-					s := strings.Split(directives[dir][0], ",")
-					l = len(s)
-					if l != 1 {
-						directives[dir] = s
-					}
-				} else if (l == 0 || c == nil) && def != "" {
-					s := strings.Split(def, ",")
-					l = len(s)
-					directives[dir] = s
-				}
-				v := reflect.MakeSlice(typeField.Type, l, l)
-				valueField.Set(v)
-				t := valueField.Interface().([]string)
-				for i, val := range directives[dir] {
+				for i, val := range value {
 					t[i] = strings.TrimSpace(val)
 				}
 			}
 		}
 	}
+
+	return nil
+}
+
+// Parser parses configuration found in the file with the specified path.
+func Parser(filepath string, f interface{}) error {
+	var reader io.Reader
+
+	if filepath != "" {
```

Comment:
Also after searching for usages, this is usually `buildcfg.SINGULARITY_CONF_FILE`. Can it be empty?

Reply:
For explicitness we can do smth like
```
if filepath == "" {
   makeDefault(f)
   return nil 
}
```

so that we don't parse regex and do all other checks.

Reply:
Will look